examples/active_plasma_lens/analysis_APL_ChrPlasmaLens_defocusing.py

```diff
@@ -0,0 +1,80 @@
+#!/usr/bin/env python3
+#
+# Copyright 2022-2025 ImpactX contributors
+# Authors: Axel Huebl, Chad Mitchell, Kyrre Sjobak
+# License: BSD-3-Clause-LBNL
+#
+# -*- coding: utf-8 -*-
+
+import numpy as np
+from analysis_APL_ChrPlasmaLens import get_beams, get_moments
+
+# initial/final beam
+(initial, beam_final, final) = get_beams()
+
+# compare number of particles
+num_particles = 10000
+assert num_particles == len(initial)
+assert num_particles == len(final)
+
+print("Initial Beam:")
+sigx, sigy, sigt, emittance_x, emittance_y, emittance_t = get_moments(initial)
+print(f"  sigx={sigx:e} sigy={sigy:e} sigt={sigt:e}")
+print(
+    f"  emittance_x={emittance_x:e} emittance_y={emittance_y:e} emittance_t={emittance_t:e}"
+)
+
+atol = 0.0  # ignored
+rtol = 2.2 * num_particles**-0.5  # from random sampling of a smooth distribution
+print(f"  rtol={rtol} (ignored: atol~={atol})")
+
+# Compare to analytical values
+assert np.allclose(
+    [sigx, sigy, sigt, emittance_x, emittance_y, emittance_t],
+    [
+        0.00010003246877770656,
+        0.00010003246877770656,
+        0.001,
+        2.548491664266332e-08,
+        2.548491664266332e-08,
+        1e-06,
+    ],
+    rtol=rtol,
+    atol=atol,
+)
+
+
+print("")
+print("Final Beam:")
+sigx, sigy, sigt, emittance_x, emittance_y, emittance_t = get_moments(final)
+s_ref = beam_final.get_attribute("s_ref")
+gamma_ref = beam_final.get_attribute("gamma_ref")
+print(f"  sigx={sigx:e} sigy={sigy:e} sigt={sigt:e}")
+print(
+    f"  emittance_x={emittance_x:e} emittance_y={emittance_y:e} emittance_t={emittance_t:e}\n"
+    f"  s_ref={s_ref:e} gamma_ref={gamma_ref:e}"
+)
+
+atol = 0.0  # ignored
+# rtol = 2.2 * num_particles**-0.5  # from random sampling of a smooth distribution
```

Comment:
Where does this 2.2 actually come from? For some reason, the sample sigma seems to always start a bit high, and with defocusing this is blown up to *just* over the limit.

I relaxed the constrain to let it pass, but I'm not sure if it's the right thing to do.

<img width="1238" height="918" alt="image" src="https://github.com/user-attachments/assets/7483651d-209e-450d-9f93-7c0a3751d0a3" />

Reply:
Exactly. Please feel free to slighly relax the constant from 2.2 to a higher number if need be.